web/pdf_link_service.js

```diff
@@ -258,9 +258,15 @@ var PDFLinkService = (function PDFLinkServiceClosure() {
             mode: params.pagemode
           });
         }
-      } else if (isPageNumber(hash)) { // Page number.
-        this.page = hash | 0;
       } else { // Named (or explicit) destination.
+        if ((typeof PDFJSDev === 'undefined' || PDFJSDev.test('GENERIC')) &&
+            (isPageNumber(hash) && hash < this.pagesCount)) {
```

Comment:
This should be `hash <= this.pagesCount`. Try with the Tracemonkey paper: #13 will give the warning, #14 will not, even though it's a valid page number.

Moreover, we can get rid of the extra parentheses around the second condition.